### PR TITLE
Creating tables

### PR DESCRIPTION
**Pull request by KacFlor**

### CODE REVIEW

**Comment by Ficzer** [comment]
Branches should not be named "review", derive branch name from task, for example "feature-#1-create-object-model"

File: src/main/java/com/example/demo/Carddata/Carddata.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.example.demo.Carddata;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Carddata/Carddata.java:1]: Branches should not be named "review", derive branch name from task, for example "feature-1-object-model"

File: src/main/java/com/example/demo/Carddata/Carddata.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import jakarta.persistence.Entity;
+
+import jakarta.persistence.Table;
+
+@Entity
+@Table
+public class Carddata {
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Carddata/Carddata.java:9]: CardData

File: src/main/java/com/example/demo/Carddata/Carddata.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    private Integer id;
+    private Double Card_num;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Carddata/Carddata.java:11]: Let's stick to camel case, I dont know this style, apply camel case everywhere

File: src/main/java/com/example/demo/Carddata/Carddata.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    public Carddata() {
+    }
+
+    public Carddata(Integer id, Double card_num) {
+        this.id = id;
+        Card_num = card_num;
+    }
+
+    public Integer getId() {
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Carddata/Carddata.java:21]: Remove this getter and setters, use Lombok annotations for automatic getter and setter generation

File: src/main/java/com/example/demo/Carddata/Carddata.java (hunk 2 above)
Comment (Ficzer) [reply src/main/java/com/example/demo/Carddata/Carddata.java:9]: Change file name also

File: src/main/java/com/example/demo/Carddata/Carddata.java (hunk 3 above)
Comment (Ficzer) [reply src/main/java/com/example/demo/Carddata/Carddata.java:11]: Card number as Double??? Isn't card number something like this "1234 5678 9123 4567"

File: src/main/java/com/example/demo/Cart/Cart.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    public Cart() {
+    }
+
+    public Cart(Integer id, Iterable quantity) {
+        this.id = id;
+        this.quantity = quantity;
+    }
+
+    public Integer getId() {
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Cart/Cart.java:20]: Lombok

File: src/main/java/com/example/demo/Cart/Cart.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
+package com.example.demo.Cart;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+@Entity
+@Table
+public class Cart {
+    private Integer id;
+    private Iterable quantity;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Cart/Cart.java:10]: Shouldn't cart has multiple products, as far as i can see there is an error in table uml

File: src/main/java/com/example/demo/Carddata/CarddataService.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+@Service
+public class CarddataService {
+
+    @Autowired
+    public CarddataService() {
+
+    }
+
+    public List<Carddata> getCarddata() {return List.of();}
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Carddata/CarddataService.java:16]: Where is DataRepository? Will it be added later?

File: src/main/java/com/example/demo/Category/Category.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
+package com.example.demo.Category;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+@Entity
+@Table
+public class Category {
+    private Integer id;
+    private  String name;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Category/Category.java:10]: extra space

File: src/main/java/com/example/demo/Customer/Customer.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+@Entity
+@Table
+public class Customer {
+    private Integer id;
+    private  String first_name;
+    private  String last_name;
+    private  String email;
+    private  String address;
+    private  Number phone_number;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Customer/Customer.java:14]: Where are the lists of orders, shipments itp? And CardData

File: src/main/java/com/example/demo/Customer/Customer.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
+package com.example.demo.Customer;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+@Entity
+@Table
+public class Customer {
+    private Integer id;
+    private  String first_name;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Customer/Customer.java:10]: extra whitespace

**Review by Ficzer — COMMENTED** [review]
These comments apply to the whole code, remember to fix everything, not only where I have added the comments

File: src/main/java/com/example/demo/Carddata/Carddata.java (hunk 1 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Carddata/Carddata.java:1]: done

File: src/main/java/com/example/demo/Carddata/Carddata.java (hunk 2 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Carddata/Carddata.java:9]: done

File: src/main/java/com/example/demo/Carddata/Carddata.java (hunk 3 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Carddata/Carddata.java:11]: done

File: src/main/java/com/example/demo/Carddata/Carddata.java (hunk 4 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Carddata/Carddata.java:21]: done

File: src/main/java/com/example/demo/Cart/Cart.java (hunk 5 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Cart/Cart.java:20]: done

File: src/main/java/com/example/demo/Cart/Cart.java (hunk 6 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Cart/Cart.java:10]: done

File: src/main/java/com/example/demo/Carddata/CarddataService.java (hunk 7 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Carddata/CarddataService.java:16]: done

File: src/main/java/com/example/demo/Category/Category.java (hunk 8 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Category/Category.java:10]: done

File: src/main/java/com/example/demo/Customer/Customer.java (hunk 10 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Customer/Customer.java:10]: done

File: src/main/java/com/example/demo/Customer/Customer.java (hunk 9 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Customer/Customer.java:14]: done (shipment and customer are in one to one relation)

**Pull request closed without merging**